dev/generate_artifacts.jl

```diff
@@ -0,0 +1,82 @@
+"2008g", "2008h", "2008i", "2009a", "2009b", "2009c", "2009d", "2009e", "2009f", "2009g", "2009h", "2009i", "2009j",
+"2009k", "2009l", "2009m", "2009n", "2009o", "2009p", "2009q", "2009r", "2009s", "2009t", "2009u", "2010a", "2010b",
+"2010c", "2010d", "2010e", "2010f", "2010g", "2010h", "2010i", "2010j", "2010k", "2010l", "2010m", "2010n", "2010o",
+"2011a", "2011b", "2011c", "2011d", "2011e", "2011f", "2011g", "2011h", "2011i", "2011j", "2011k", "2011l", "2011m",
+"2011n", "2012a", "2012b", "2012c", "2012d", "2012e", "2012f", "2012g", "2012h", "2012i", "2012j", "2013a", "2013b",
+"2013c", "2013d", "2013e", "2013f", "2013g", "2013h", "2013i", "2014a", "2014b", "2014c", "2014d", "2014e", "2014f",
+"2014g", "2014h", "2014i", "2014j", "2015a", "2015b", "2015c", "2015d", "2015e", "2015f", "2015g", "2016a", "2016b",
+"2016c", "2016d", "2016e", "2016f", "2016g", "2016h", "2016i", "2016j", "2017a", "2017b", "2017c", "2018a", "2018b",
+"2018c", "2018d", "2018e", "2018f", "2018g", "2018h", "2018i", "2019a", "2019b", "2019c", "2020a", "93g", "94a", "94b",
+"94d", "94e", "94f", "94h", "95b", "95c", "95d", "95e", "95f", "95g", "95h", "95i", "95k", "95l", "95m", "96a", "96b",
+"96c", "96d", "96e", "96h", "96i", "96k"]
+
+for version in versions
+    # Query the `Artifacts.toml` file for the hash bound to the specific version
+    # (returns `nothing` if no such binding exists)
+    tzarchive_latest_hash = artifact_hash("tzdata_$version", artifacts_toml)
+
+    # If the name was not bound, or the hash it was bound to does not exist, create it!
+    if isnothing(tzarchive_latest_hash) || !artifact_exists(tzarchive_latest_hash)
+        # create_artifact() returns the content-hash of the artifact directory once we're finished creating it
+        tzfile_hash = create_artifact() do artifact_dir
+            # We create the artifact by simply downloading a few files into the new artifact directory
+            @info "Downloading $version tzdata"
+            archive_name = tzdata_download(version, artifact_dir)
+            extract(archive_name, artifact_dir)
+            rm(archive_name)
+        end
+        download_dir = artifact_path(tzfile_hash)
+        content_sha = open(tzdata_download(version)) do f
+           bytes2hex(sha256(f))
+        end
+
+        # Now bind that hash within our `Artifacts.toml`.  `force = true` means that if it already exists,
+        # just overwrite with the new content-hash.  Unless the source files change, we do not expect
+        # the content hash to change, so this should not cause unnecessary version control churn.
+        download_data = [(tzdata_url(version), content_sha)]
+        bind_artifact!(artifacts_toml, "tzdata_$version", tzfile_hash, lazy=true, download_info=download_data, platform=platform_key_abi())
+    end
+end
+
+# this can not work in current state because Artifacts require tarball, and there is only the direct XML
+# # and now, add windows xml thingy
```

Comment:
If Artifacts can handle zip files then you can use `https://github.com/unicode-org/cldr/archive/<tag>.zip` to get a copy of the entire CLDR repo

Reply:
Oh, that's true, I didn't want to download the whole repo because of single xml, but I guess it will be the best current solution.

Reply:
Did you open an issue against Pkg for allowing non-archive downloads?

Reply:
No, should I?

Reply:
I think this is use case that is reasonable and an I think an issue is warranted

Reply:
So, I made an issue, we'll see. https://github.com/JuliaLang/Pkg.jl/issues/1921